### DISCUSSION TITLE
gmail server taking 2 minutes to open connection.

### DISCUSSION THREAD

**Discussion by humblewizard** [PHPMailer/PHPMailer#2606] [General]
Gmail is suddenly taking a long time to open a connection. Here is the SMTPDebug = 3 output:

2022-01-13 20:10:18 Connection: opening to smtp.gmail.com:587, timeout=300, options=array()
2022-01-13 20:12:27 Connection: opened

Is gmail playing games now with the simple log in or am I doing something wrong?

Here is the setup:

                $mail->SMTPAuth   = true;
                $mail->isSMTP();
                $mail->Sender = "*************@yachtdestinations.org";
                $mail->SMTPSecure = "tls";                 // sets the prefix to the server
                $mail->Host       = "smtp.gmail.com";      // sets GMAIL as the SMTP server
                $mail->Port       = 587;                   // set the SMTP port
                $mail->Username   = "***********@yachtdestinations.org";  // GMAIL username
                $mail->Password   = "***********";

**Comment by Synchro** [comment]
This is just something you should be prepared to deal with. SMTP has very long timeouts that make using it in interactive situations (e.g. web page submission processing) very risky. The best way to deal with this problem is to run a local mail server (which will have no such delays) as a relay.

**Reply by humblewizard** [reply]
Will using a local mail server have reputation consequences?

**Reply by Synchro** [reply]
No, because you can set it up to relay through whatever mail server you were originally using, in your case gmail.

**Reply by humblewizard** [reply]
OK, I think I'll try that. Any recommendations on a how to guide? I'm looking at:

https://www.howtoforge.com/tutorial/configure-postfix-to-use-gmail-as-a-mail-relay/

**Reply by Synchro** [reply]
Looks good, though bear in mind you may run into the same issues with authentication on gmail as PHPMailer does, e.g. needing to do the unlock captcha thing, configure XOAUTH2, see the troubleshooting guide for more info.
